platform/v05/exit.go

```diff
@@ -0,0 +1,31 @@
+package v05
+
+var exitCodes = map[string]int{
+	// detect phase errors: 100-199
+	"FailedDetect":           100, // FailedDetect indicates that no buildpacks detected
+	"FailedDetectWithErrors": 101, // FailedDetectWithErrors indicated that no buildpacks detected and at least one errored
+	"DetectError":            102, // DetectError indicates generic detect error
+
+	// analyze phase errors: 200-299
+	"AnalyzeError": 202, // AnalyzeError indicates generic analyze error
+
+	// restore phase errors: 300-399
+	"RestoreError": 302, // RestoreError indicates generic restore error
+
+	// build phase errors: 400-499
+	"FailedBuildWithErrors": 401, // FailedBuildWithErrors indicates buildpack error during /bin/build
+	"BuildError":            402, // BuildError indicates generic build error
+
+	// export phase errors: 500-599
+	"ExportError": 502, // ExportError indicates generic export error
+
+	// rebase phase errors: 600-699
+	"RebaseError": 602, // RebaseError indicates generic rebase error
+
+	// launch phase errors: 700-799
+	"LaunchError": 702, // LaunchError indicates generic launch error
+}
+
+func (p *Platform) CodeFor(errType string) int {
+	return exitCodes[errType]
```

Comment:
What will happen when we want to introduce new exit codes?
In the current implementation, we should probably need to add them to both api < 0.6 and to api >=0.5.
Of course that theoretically, we shouldn't get to this function with new codes but I think that with the current implementation, it's better to do it so we won't get a bad error for searching a key that doesn't exist in the map.
Since I prefer not to add new codes to old apis, I think we should change this function to check first if the key exists in the map and if not, return a general error.
Moreover, I don't like so much the fact that `errType` is a string. I feel that contributors might make mistake when writing them.
I'm not sure what is the best way to make this change but what do you think about having a common file for all apis that will contain a list of constants and we'll use it everywhere?

Reply:
+1 to checking that the code exists.

I agree that keying on a string feels weird. What would the file of constants look like?

Reply:
I thought about using enums but after searching in Google, the implementation of enums in golang is a bit different.
Here are some links that I found and might be useful:
https://www.educative.io/edpresso/what-is-an-enum-in-golang
https://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-go

Reply:
Thank you for the suggestion! Fixed in 8faa92e and bf07a2a.

Reply:
I haven't checked our lint rules, but I know there are configurable lint rules for ensuring exhaustive switch statements with enums. Maybe removing the map in favor of a switch in each version's `exit.go` would work and alert us to handling new errors as they are introduced?